packages/compartment-mapper/test/test-integrity.js

```diff
@@ -0,0 +1,78 @@
+// @ts-check
+import 'ses';
+import test from 'ava';
+import { ZipReader, ZipWriter } from '@endo/zip';
+import { makeArchive, parseArchive } from '../index.js';
+import { readPowers } from './scaffold.js';
+
+const fixture = new URL(
+  'fixtures-0/node_modules/app/main.js',
+  import.meta.url,
+).toString();
+
+test('extracting an archive with a superfluous file', async t => {
+  const validBytes = await makeArchive(readPowers, fixture, {
+    modules: {
+      builtin: null,
+    },
+    dev: true,
+  });
+
+  const reader = new ZipReader(validBytes);
+  const writer = new ZipWriter();
+  writer.files = reader.files;
+  writer.write('extraneous.txt', new TextEncoder().encode('Extra'));
+  const invalidBytes = writer.snapshot();
+
+  await t.throwsAsync(
+    () =>
+      parseArchive(invalidBytes, 'extra.zip', {
+        computeSha512: readPowers.computeSha512,
+        modules: {
+          builtin: null,
+        },
+      }),
+    {
+      message:
+        'Archive contains extraneous files: ["extraneous.txt"] in "extra.zip"',
+    },
+  );
+
+  t.pass();
+});
+
+test('extracting an archive with an inconsistent hash', async t => {
```

Comment:
This tests that a module/member file can't be corrupted.. do we already have a test that the compartment map cannot be corrupted in a similar way?

Also, is there a test that a module/member file cannot be missing? If the resulting behavior is that the module cannot be loaded, that's fail-safe, but I do want the error to be sensed early (e.g. when swingset is asking the question "does this bundle match this ID?"), rather than not appearing until we try to evaluate code.

Reply:
I’ve added two compartment map corruption tests. One compares the valid and invalid returned hashes. The other makes use of the `expectedSha512` option on `parseArchive`. I’ve also added a test for a missing file.